### PR TITLE
Add support of selective mapping to backend console

### PR DESCRIPTION
**Pull request by jasiu001**
Related to #3843 

- graphQL implementation for selective mapping

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
Please, add an endpoint which provides information about the mapping (query)

File: components/console-backend-service/internal/gqlschema/schema.graphql [hunk 1]
```diff
@@ -606,16 +608,31 @@ type Application {
     labels: Labels!
     services: [ApplicationService!]!
     enabledInNamespaces: [String!]!
+    enabledMappingServices: [enabledMappingService]
     status: ApplicationStatus!
 }
 
+type enabledMappingService {
+    namespace: String!
+    allServices: Boolean
```
Comment (polskikiel) [inline components/console-backend-service/internal/gqlschema/schema.graphql]: we can make `allServices` mandatory

File: components/console-backend-service/internal/domain/application/app_resolver.go [hunk 2]
```diff
@@ -170,9 +172,62 @@ func (r *applicationResolver) EnableApplicationMutation(ctx context.Context, app
...
+
+	return &gqlschema.ApplicationMapping{
+		Namespace:   em.Namespace,
+		Application: em.Name,
+		AllServices: allServices,
+		Services:    r.transformApplicationMappingService(em.Spec.Services),
 	}, nil
 }
 
+func configureServices(allServices *bool, services []*gqlschema.ApplicationMappingService) []*gqlschema.ApplicationMappingService {
```
Comment (polskikiel) [inline components/console-backend-service/internal/domain/application/app_resolver.go]: You can move that function to service or just make it one of the `applicationResolver` method:
`(r *applicationResolver)`

File: components/console-backend-service/internal/gqlschema/schema.graphql [hunk 3]
```diff
@@ -890,7 +907,8 @@ type Mutation {
...
     deleteApplication(name: String!): DeleteApplicationOutput! @HasAccess(attributes: {resource: "applications", verb: "delete", apiGroup: "applicationconnector.kyma-project.io", apiVersion: "v1alpha1", nameArg: "name"})
 
-    enableApplication(application: String!, namespace: String!): ApplicationMapping @HasAccess(attributes: {resource: "applicationmappings", verb: "create", apiGroup: "applicationconnector.kyma-project.io", apiVersion: "v1alpha1", namespaceArg: "namespace"})
+    enableApplication(application: String!, namespace: String!, allServices: Boolean, services: [ApplicationMappingService]): ApplicationMapping @HasAccess(attributes: {resource: "applicationmappings", verb: "create", apiGroup: "applicationconnector.kyma-project.io", apiVersion: "v1alpha1", namespaceArg: "namespace"})
+    overloadApplication(application: String!, namespace: String!, allServices: Boolean, services: [ApplicationMappingService]): ApplicationMapping @HasAccess(attributes: {resource: "applicationmappings", verb: "create", apiGroup: "applicationconnector.kyma-project.io", apiVersion: "v1alpha1", namespaceArg: "namespace"})
```
Comment (polskikiel) [inline components/console-backend-service/internal/gqlschema/schema.graphql]:
```suggestion
    overloadApplication(application: String!, namespace: String!, allServices: Boolean, services: [ApplicationMappingService]): ApplicationMapping @HasAccess(attributes: {resource: "applicationmappings", verb: "update", apiGroup: "applicationconnector.kyma-project.io", apiVersion: "v1alpha1", namespaceArg: "namespace"})
```

File: components/console-backend-service/internal/domain/application/app_resolver.go [hunk 4]
```diff
@@ -200,6 +255,65 @@ func (r *applicationResolver) ApplicationEnabledInNamespacesField(ctx context.Co
...
+	}
+
+	var collection []*gqlschema.EnabledMappingService
+	for _, mapping := range items {
+		ems := &gqlschema.EnabledMappingService{}
+
+		ems.Namespace = mapping.Namespace
+		all := mapping.IsAllApplicationServicesEnabled()
+		ems.AllServices = &all
+		ems.Services = findEnabledServices(obj.Services, mapping)
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply components/console-backend-service/internal/domain/application/app_resolver.go]: I prefer to inject as less argument as possible to functions. I changed this part of code in different way, method `IsAllApplicationServicesEnabled` is used only once.

File: components/console-backend-service/internal/domain/application/app_resolver.go [hunk 5]
```diff
@@ -200,6 +255,65 @@ func (r *applicationResolver) ApplicationEnabledInNamespacesField(ctx context.Co
...
+		for _, val := range appServices {
+			if val.ID == id {
+				return val.DisplayName, true
+			}
+		}
+
+		return "", false
+	}
+
+	var result []*gqlschema.EnabledService
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply components/console-backend-service/internal/domain/application/app_resolver.go]: struct `EnabledMappingService` already exist, is a parent of `EnabledService`

**Pull request merged**